From 290e505b287fe664549cb72b26507d8ffb4a6d39 Mon Sep 17 00:00:00 2001
From: Vadim Pasternak <vadimp@mellanox.com>
Date: Mon, 2 Oct 2017 15:13:05 +0000
Subject: [v4.9 backport 29/29] platform/x86: mlx-platform: add support for new
 system types

It adds support for new Mellanox system types: SN3700 (32x200GbE and
16x400GbE Ethernet switch), SN3410 (6x400GbE plus 48x50GbE Ethernet
switch).

Signed-off-by: Vadim Pasternak <vadimp@mellanox.com>
---
 drivers/platform/x86/mlx-platform.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/drivers/platform/x86/mlx-platform.c b/drivers/platform/x86/mlx-platform.c
index d7d8fca..aba084c 100644
--- a/drivers/platform/x86/mlx-platform.c
+++ b/drivers/platform/x86/mlx-platform.c
@@ -713,6 +713,20 @@ static struct dmi_system_id mlxplat_dmi_table[] __initdata = {
 			DMI_MATCH(DMI_PRODUCT_NAME, "QMB7"),
 		},
 	},
+	{
+		.callback = mlxplat_dmi_qmb7xx_matched,
+		.matches = {
+			DMI_MATCH(DMI_BOARD_VENDOR, "Mellanox Technologies"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "SN37"),
+		},
+	},
+	{
+		.callback = mlxplat_dmi_qmb7xx_matched,
+		.matches = {
+			DMI_MATCH(DMI_BOARD_VENDOR, "Mellanox Technologies"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "SN34"),
+		},
+	},
 	{ }
 };
 
@@ -832,3 +846,5 @@ MODULE_ALIAS("dmi:*:*Mellanox*:MSN21*:");
 MODULE_ALIAS("dmi:*:*Mellanox*MSN274*:");
 MODULE_ALIAS("dmi:*:*Mellanox*MSN201*:");
 MODULE_ALIAS("dmi:*:*Mellanox*QMB7*:");
+MODULE_ALIAS("dmi:*:*Mellanox*SN37*:");
+MODULE_ALIAS("dmi:*:*Mellanox*QM34*:");
-- 
2.1.4

